$NetBSD: patch-an,v 1.1 2008/07/24 01:07:31 bjs Exp $

Fix missing error condition (from GIT)

added error check in Xcms color file parser; closes bug #15305

--- src/xcms/cmsColNm.c.orig	2008-03-06 15:45:06.000000000 -0500
+++ src/xcms/cmsColNm.c
@@ -735,10 +735,17 @@ LoadColornameDB(void)
     }
 
     if ((stream = _XFopenFile (pathname, "r")) == NULL) {
+	/* can't open file */
+	XcmsColorDbState = XcmsDbInitFailure;
 	return(XcmsFailure);
     }
 
-    stringSectionSize(stream, &nEntries, &size);
+    if (stringSectionSize(stream, &nEntries, &size) != XcmsSuccess ||
+	nEntries == 0) {
+	(void) fclose(stream);
+	XcmsColorDbState = XcmsDbInitFailure;
+	return(XcmsFailure);
+    }
     rewind(stream);
 
     strings = (char *) Xmalloc(size);
